core/src/main/java/edu/wpi/grip/core/sockets/InputSocket.java

```diff
@@ -114,7 +114,14 @@ public void onValueChanged() {
 
     @Override
     public boolean equals(Object o) {
-      return o instanceof InputSocket && decorated.equals(o);
+      if (o instanceof InputSocket) {
+        if (decorated.equals(o)) {
+          return true;
+        } else if (o instanceof Decorator) {
+          return decorated.equals(((Decorator) o).decorated);
+        }
```

Comment:
Yes but unless GRIP has a style saying "we prefer single line equals methods", multiple lines is easier to read and less prone to accidental errors.